libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/DirectionsSession.kt

```diff
@@ -17,6 +18,10 @@ internal interface DirectionsSession {
 
     fun unregisterRoutesObserver(routesObserver: RoutesObserver)
 
+    fun registerFasterRouteObserver(fasterRouteObserver: FasterRouteObserver)
```

Comment:
As mentioned in https://github.com/mapbox/mapbox-navigation-android/pull/2462, if `MapboxNavigation` is going to be responsible for making the decision if the route is faster, it could also notify the users back and maintain the observers' list. We don't need to return the control back to the `DirectionsSession`.

Otherwise, I'd prefer `MapboxDirectionsSession` to have a constructor injected `FasterRouteDetector` from https://github.com/mapbox/mapbox-navigation-android/pull/2462 instead of the `FasterRouteRequestCallback` mechanism. We ultimately don't need to worry about whether or not the requested succeeded as the session's client.